### PR TITLE
[IE][VPU][GT] Add pass to reshape convolution by parameter from IR

### PR DESCRIPTION
**Pull request by ZlobinGM**
This pass reads parameter named "ConvReshape". Then reshape convolution using parameter value. 
Reshape only convolutions with kernel 1x1 and only by height and width.

### CODE REVIEW

File: inference-engine/src/readers/ir_reader/ie_ir_parser.cpp [hunk 1]
```diff
@@ -569,6 +569,10 @@ std::shared_ptr<ngraph::Node> V10Parser::createNode(const std::vector<ngraph::Ou
         if (pr_data) {
             rtInfo["PrimitivesPriority"] = std::make_shared<::ngraph::VariantWrapper<std::string> >(pr_data.value());
         }
+        const auto cr_data = dn.attribute("ConvReshape");
```
Comment (ilyachur) [inline inference-engine/src/readers/ir_reader/ie_ir_parser.cpp:572]: Why do we need to support this attribute in the parser?

Who should define this runtime information? If user should do it, should we describe this runtime attribute to the documentation?

File: inference-engine/src/vpu/graph_transformer/src/middleend/passes/reshape_conv.cpp [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+        std::string name = stage->name();
+        int inputC = inputDesc.dim(Dim::C);
+        int outputC = outputDesc.dim(Dim::C);
+        int dimH = inputDesc.dim(Dim::H);
+        int dimW = inputDesc.dim(Dim::W);
+        int resultW = 0, resultH = 0;
+        resultH = ChoiceDimH(name, inputC, outputC, dimH, dimW);
+
+        if (stage->origLayer()->params.count("ConvReshape")) {
```
Comment (andrejsokolov) [inline inference-engine/src/vpu/graph_transformer/src/middleend/passes/reshape_conv.cpp:60]: count("ConvReshape") should be equal to 1?

Comment (ZlobinGM) [reply inference-engine/src/vpu/graph_transformer/src/middleend/passes/reshape_conv.cpp:63]: count("ConvReshape") returns 0 or 1, so implicit conversion works good here

File: inference-engine/src/vpu/graph_transformer/src/middleend/passes/reshape_conv.cpp [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    explicit PassImpl(const StageBuilder::Ptr& stageBuilder) : _stageBuilder(stageBuilder) {}
+
+    void run(const Model& model) override;
+
+private:
+    StageBuilder::Ptr _stageBuilder;
+};
+
+void PassImpl::run(const Model& model) {
+    VPU_PROFILE(hwConvTiling);
```
[the remark this answers is not in the document]
Comment (ZlobinGM) [reply inference-engine/src/vpu/graph_transformer/src/middleend/passes/reshape_conv.cpp]: Yes. Fixed

File: inference-engine/src/vpu/graph_transformer/src/middleend/passes/reshape_conv.cpp [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            stage->attrs().get<int>("kernelSizeY") != 1)
+            continue;
+
+        std::string name = stage->name();
+        int inputC = inputDesc.dim(Dim::C);
+        int outputC = outputDesc.dim(Dim::C);
+        int dimH = inputDesc.dim(Dim::H);
+        int dimW = inputDesc.dim(Dim::W);
+        int resultW = 0, resultH = 0;
+        resultH = ChoiceDimH(name, inputC, outputC, dimH, dimW);
```
[the remark this answers is not in the document]
Comment (ZlobinGM) [reply inference-engine/src/vpu/graph_transformer/src/middleend/passes/reshape_conv.cpp]: fixed

File: inference-engine/src/readers/ir_reader/ie_ir_parser.cpp (hunk 1 above)
Comment (ZlobinGM) [reply inference-engine/src/readers/ir_reader/ie_ir_parser.cpp:572]: @taka-no-me , could you please comment it?

File: inference-engine/src/vpu/graph_transformer/src/middleend/passes/reshape_conv.cpp [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+            } catch (...) {
+                resultW = resultH = 0;
+            }
+        }
+
+        if (resultH == 0) {
+            continue;
+        }
+
+        resultW = dimH * dimW / resultH;
```
Comment (andrejsokolov) [inline inference-engine/src/vpu/graph_transformer/src/middleend/passes/reshape_conv.cpp:77]: should we add a check to make sure that dimH * dimW = resultH * resultW ?

**Review by andrejsokolov — APPROVED** [review]

File: inference-engine/src/vpu/graph_transformer/src/middleend/passes/reshape_conv.cpp (hunk 5 above)
Comment (ZlobinGM) [reply inference-engine/src/vpu/graph_transformer/src/middleend/passes/reshape_conv.cpp]: Added IE_ASSERT

File: inference-engine/src/vpu/graph_transformer/src/middleend/passes/reshape_conv.cpp [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+        int dimW = inputDesc.dim(Dim::W);
+        int resultH = 0;
+        int resultW = 0;
+
+        resultH = choiceDimH(name, inputC, outputC, dimH, dimW);
+
+        if (stage->origLayer()->params.count("alt_width")) {
+            std::string rtParam = stage->origLayer()->params.at("alt_width");
+            try {
+                resultW = std::stoi(rtParam);
```
[the remark this answers is not in the document]
Comment (ZlobinGM) [reply inference-engine/src/vpu/graph_transformer/src/middleend/passes/reshape_conv.cpp]: Added check for number

File: inference-engine/src/readers/ir_reader/ie_ir_parser.cpp [hunk 7]
```diff
@@ -559,6 +559,10 @@ std::shared_ptr<ngraph::Node> V10Parser::createNode(const std::vector<ngraph::Ou
         if (pr_data) {
             rtInfo["PrimitivesPriority"] = std::make_shared<::ngraph::VariantWrapper<std::string> >(pr_data.value());
         }
+        const auto cr_data = dn.attribute("alt_width");
```
[the remark this answers is not in the document]
Comment (ZlobinGM) [reply inference-engine/src/readers/ir_reader/ie_ir_parser.cpp]: Rename to aw_data (a - alt, w - width)

File: inference-engine/src/vpu/graph_transformer/include/vpu/middleend/hw/conv_tiling/reshape_conv_func.hpp [hunk 8]
```diff
@@ -0,0 +1,9 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+// Function for custom reshape condition for conv 1x1
+
+int choiceDimW(std::string name, int inC, int outC, int dimH, int dimW) {
```
Comment (gladilov-gleb) [inline inference-engine/src/vpu/graph_transformer/include/vpu/middleend/hw/conv_tiling/reshape_conv_func.hpp:7]: Still, why it just returns 0?

File: inference-engine/src/vpu/graph_transformer/include/vpu/middleend/hw/conv_tiling/reshape_conv_func.hpp [hunk 9]
```diff
@@ -0,0 +1,9 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+// Function for custom reshape condition for conv 1x1
+
+int choiceDimH(std::string name, int inC, int outC, int dimH, int dimW) {
```
[the remark this answers is not in the document]
Comment (ZlobinGM) [reply inference-engine/src/vpu/graph_transformer/include/vpu/middleend/hw/conv_tiling/reshape_conv_func.hpp]: This function is user filled.

**Review by gladilov-gleb — APPROVED** [review]

**Pull request merged**